controllers/backup.go

```diff
@@ -17,8 +18,13 @@ func (r *DirectoryServiceReconciler) updateBackup(ctx context.Context, ds *direc
 
 	if !ds.Spec.Backup.Enabled {
 		log.V(5).Info("Backup is disabled. Nothing to do")
-		// Delete the backup tasks - even if they dont exist - this is no more expensive than querying then deleting..
-		l.DeleteBackupTask(ds.Name)
+		bStatus, err := l.GetBackupTaskStatus(ds.Name)
```

Comment:
This reduces `"FAILED"` error messages in DS logs. We were trying to delete the Tasks without checking if they were there.